src/core/lib/channel/metrics.h

```diff
@@ -288,7 +478,7 @@ class GlobalStatsPluginRegistry {
     GRPC_MUST_USE_RESULT std::unique_ptr<RegisteredMetricCallback>
     RegisterCallback(
         absl::AnyInvocable<void(CallbackMetricReporter&)> callback,
-        std::vector<GlobalInstrumentsRegistry::GlobalCallbackHandle> metrics,
+        std::vector<GlobalInstrumentsRegistry::GlobalInstrumentHandle> metrics,
```

Comment:
This change actually loses us a bit of type safety: previously, it was not possible to register a callback for a metric with InstrumentType != kCallbackGauge, but this change seems to allow it for other types as well.

Is there some reasonable way we can get type safety back for this?

Reply:
Great point! I changed this API a bit to be able to statically check the handles' type. PTAL.